grin/src/sync.rs

```diff
@@ -99,18 +141,33 @@ impl Syncer {
 		// check if we have missing full blocks for which we already have a header
 		self.init_download()?;
 
-		// main syncing loop, requests more headers and bodies periodically as long
+	//Keep track of which peer we're currently syncing with
+	let current_sync_buddy = None; 
+
+	//keep a map of the latest header each peer has given us
+	//(so we can swap from one to the next while still
+	//providing a relevant set of locators
+	//let last_headers_from_peers = HashMap::new();
+
 		// as a peer with higher difficulty exists and we're not fully caught up
 		info!(LOGGER, "Sync: Starting loop.");
 		loop {
 			let tip = self.chain.get_header_head()?;
 
-			// TODO do something better (like trying to get more) if we lose peers
-			let peer = self.p2p.most_work_peer().expect("No peers available for sync.");
-			let peer = peer.read().unwrap();
+			// select the best peer to be peering with
+			let current_sync_buddy=self.select_buddy(current_sync_buddy.clone());
```

Comment:
Sorry can't help it, but I think it's `<space>=<space>`. It's silly but it's been bothering me for too long :-S

Reply:
I must be .... corrrrrrrected, sir.